### PR TITLE
Remove classes from checkers to avoid serialziation bugs

### PR DESCRIPTION
**Pull request by flamecoals**
#### Checklist

* [x] Use a separate branch in your local repo (not `master`).
* [x] I've read and agree with the [contribution guidelines](https://www.freeboardgames.org/docs/?path=/story/documentation-contributing--page).

### CODE REVIEW

**Comment by mateusazis** [comment]
Do you prefer instead to just rollback #736 ?

**Comment by flamecoals** [comment]
I think this is better because we have more tests now...

File: web/src/games/checkers/coord.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  return a.x === b.x && a.y === b.y;
+}
+
+export function fromPosition(position: number): ICoord {
+  const x = position % 8;
+  const y = Math.floor(position / 8);
+  return { x, y };
+}
+
+export function fromICoord(other: ICoord): ICoord {
```
Comment (mateusazis) [inline web/src/games/checkers/coord.ts:36]: This shouldn't be necessary anymore. It was used to convert ICoord -> Coord to use the instance methods but those are now gone.

File: web/src/games/checkers/game.ts [hunk 2]
```diff
@@ -82,18 +39,18 @@ export const INITIAL_BOARD: Piece[] = [
...
 
-const KING_DIRS: Coord[] = [new Coord(-1, 1), new Coord(1, 1), new Coord(-1, -1), new Coord(1, -1)];
+const KING_DIRS: ICoord[] = [coord(-1, 1), coord(1, 1), coord(-1, -1), coord(1, -1)];
 
 export function checkPosition(
   G: IG,
   playerID: string,
   piece: ICheckerPiece,
-  coord: Coord,
+  coord: ICoord,
```
Comment (mateusazis) [inline web/src/games/checkers/game.ts:53]: This name clashes with the "coord" function. Maybe rename the later to "createCoord"?

File: web/src/games/checkers/game.ts [hunk 3]
```diff
@@ -170,10 +127,10 @@ export function getValidMoves(G: IG, playerID: string, jumping?: ICheckerPieceWi
 }
 
 export function move(G: IG, ctx: Ctx, from: ICoord, to: ICoord): IG | string {
-  const fromCoord = new Coord(from.x, from.y);
-  const toCoord = new Coord(to.x, to.y);
-  const indexFrom = fromCoord.toIndex();
-  const indexTo = toCoord.toIndex();
+  const fromCoord = coord(from.x, from.y);
+  const toCoord = coord(to.x, to.y);
```
Comment (mateusazis) [inline web/src/games/checkers/game.ts:131]: These 2 initial conversions should also be unnecessary (see my previous comment on "fromICoord").

File: web/src/games/checkers/coord.ts (hunk 1 above)
Comment (mateusazis) [inline web/src/games/checkers/coord.ts:36]: This shouldn't be necessary anymore. It was used to convert ICoord -> Coord to use the instance methods but those are now gone.

File: web/src/games/checkers/game.ts (hunk 3 above)
Comment (flamecoals) [reply web/src/games/checkers/game.ts:131]: Done.

File: web/src/games/checkers/game.ts (hunk 2 above)
Comment (flamecoals) [reply web/src/games/checkers/game.ts:53]: Done.

File: web/src/games/checkers/coord.ts (hunk 1 above)
Comment (flamecoals) [reply web/src/games/checkers/coord.ts:36]: Done.

File: web/src/games/checkers/game.ts [hunk 4]
```diff
@@ -127,8 +127,8 @@ export function getValidMoves(G: IG, playerID: string, jumping?: ICheckerPieceWi
 }
 
 export function move(G: IG, ctx: Ctx, from: ICoord, to: ICoord): IG | string {
-  const fromCoord = coord(from.x, from.y);
-  const toCoord = coord(to.x, to.y);
+  const fromCoord = createCoord(from.x, from.y);
```
Comment (mateusazis) [inline web/src/games/checkers/game.ts:130]: Sorry, what I meat here is that you don't need the "fromCoord"and "toCoord" variables at all. You can just pass "from" and "to" directly to "toIndex" below.

**Review by mateusazis — APPROVED** [review]

File: web/src/games/checkers/game.ts (hunk 4 above)
Comment (flamecoals) [reply web/src/games/checkers/game.ts:130]: Done

**Pull request merged**